### PR TITLE
Feat/cache

### PR DESCRIPTION
**Pull request by leonardopennino**

### CODE REVIEW

**Review by adam-narozniak — COMMENTED** [review]
Hi, Leonardo.
I have a more general question. How do the PrivateStoreage and Storage relate to the db LocalRepository and Repository? Could you just briefly explain the relationship between them?

**Review by aladin6060 — APPROVED** [review]
Some tests for logging would be nice. A method to get statistics is missing. But otherwise looks good to me!

File: app/src/main/java/ch/epfl/sweng/rps/storage/PrivateStorage.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        data.put("uid", user.uid)
+        data.put("email",user.email)
+        val f = getFile(Storage.FILES.USERINFO)
+        data.store(f.bufferedWriter(), null)
+    }
+    override fun getUserSettings() {
+        TODO("Not yet implemented")
+    }
+
+    override fun getMatchesDetails() {
```
Comment (panadax) [inline app/src/main/java/ch/epfl/sweng/rps/storage/PrivateStorage.kt:48]: Actually, for stats page, we have personal stats page and for each game, we can get into its game details (all rounds of it). Should we have the local storage for it as well?

**Review by panadax — COMMENTED** [review]
LGTM, may we have more tests for it to let us know how to use your feature on our work?

**Comment by leonardopennino** [comment]
> Hi, Leonardo.
> I have a more general question. How do the PrivateStoreage and Storage relate to the db LocalRepository and Repository? Could you just briefly explain the relationship between them?

Sure! The storage interface will be a wrapper for all data related queries. The process will work as follows : 
1 A class calls  storage.getMatches()
2 Storage will return the data inside the cache if it exists (Availability over consistency). After it gets the data from the cache it will try to query the server to renew the cache.
3 Cache doesn't have the data -> Storage will ask the server to provide the data -> updates cache 
4 Returns data to the function who requested it

**Review by gaetschwartz — APPROVED** [review]

**Pull request merged**